### PR TITLE
Nits: Improve if statements

### PR DESCRIPTION
**Pull request by ajnavarro**
use `always()` instead of `(success() || failure())`

Signed-off-by: Antonio Navarro Perez <antnavper@gmail.com>

### CODE REVIEW

File: templates/.github/workflows/go-check.yml [hunk 1]
```diff
@@ -36,28 +36,28 @@ jobs:
...
+        if: always() # run this step even if the previous one failed
         uses: protocol/multiple-go-modules@v1.2
         with:
           run: |
             set -o pipefail
             staticcheck ./... | sed -e 's@\(.*\)\.go@./\1.go@g'
       - name: go generate
         uses: protocol/multiple-go-modules@v1.2
-        if: (success() || failure()) && env.RUNGOGENERATE == 'true'
+        if: env.RUNGOGENERATE == 'true'
```
Comment (marten-seemann) [inline templates/.github/workflows/go-check.yml:60]: This is wrong. This should also run if the previous step failed.

Comment (ajnavarro) [reply templates/.github/workflows/go-check.yml]: Woops, true. changed.

**Comment by galargh** [comment]
FYI, `always()` is equivalent to `success() || failure() || cancelled()` so this is in fact changing the previous behaviour.

**Comment by ajnavarro** [comment]
Feel free to close this PR if the new behavior is not correct.

**Comment by galargh** [comment]
> Feel free to close this PR if the new behavior is not correct.

Unless this is specifically requested, I think it'd be valuable to preserve the ability to cancel a workflow run and have it quit immediately. Closing for now then.

**Pull request closed without merging**